Adafruit_TCS34725.h

```diff
@@ -143,14 +143,18 @@ typedef enum {
       0xFF, /**<  2.4ms - 1 cycle    - Max Count: 1024  */
   TCS34725_INTEGRATIONTIME_24MS =
       0xF6, /**<  24ms  - 10 cycles  - Max Count: 10240 */
+  TCS34725_INTEGRATIONTIME_48MS =
```

Comment:
I believe this is incorrect, and the proper value would be 51ms, unless I'm misreading something:
```python
>>> from math import ceil
>>> for lsb in [0xFF, 0xF6, 0xEB, 0xD5, 0xC0, 0x00]:
...     cycles=(0x100-lsb)
...     int_time = cycles * 2.4
...     print("LSB:%d, cycles:%d, integration time: %.1f, delay(ms): %d"%(lsb, cycles, int_time, ceil(int_time)))
...
LSB:255, cycles:1, integration time: 2.4, delay(ms): 3
LSB:246, cycles:10, integration time: 24.0, delay(ms): 24
LSB:235, cycles:21, integration time: 50.4, delay(ms): 51
LSB:213, cycles:43, integration time: 103.2, delay(ms): 104
LSB:192, cycles:64, integration time: 153.6, delay(ms): 154
LSB:0, cycles:256, integration time: 614.4, delay(ms): 615
```
The same would apply to `TCS34725_INTEGRATIONTIME_101MS` which would be `TCS34725_INTEGRATIONTIME_104MS`.

The corresponding delay times would need to be adjusted as well.

Reply:
That's weird, the comment says 20 cycles, but the hex is not right for that.  Same with the comment saying 42 cycles but is really 43 cycles.
Sorry I was looking at the comment and should've verified the hex.

One question, do we want to stick with the hex values and change the times, comments?  Or do we want the intention of the comments preserved and change the hex value?

Reply:
Some nice times to have would be multiples of 100ms, because both 50Hz and 60Hz periods divide evenly into 100ms.

I'm not sure where some of these times came from, they don't seem useful, they just seem like they happened to be on one chart in the datasheet.

Reply:
How about adding these, they would be useful:


| hex | integ (ms) | 60Hz cycles | 50Hz cycles |
|-----|------------|-------------|-------------|
|D6   | 100.80     | 6.05        | 5.04        |
|AD   | 199.20     | 11.95       | 9.96        |
|83   | 300.00     | 18.00       | 15.00       |
|59   | 400.80     | 24.05       | 20.04       |
|30   | 499.20     | 29.95       | 24.96       |
|06   | 600.00     | 36.00       | 30.00       |

```python
for i in range(255):
    register = 256-i
    integ = i*2.4
    hz60cycles = integ*1e-3 / (1/60)
    hz50cycles = integ*1e-3 / (1/50)

    print(f"hex {register:02X}    integ {i*2.4:.2f}    60Hz cycles {hz60cycles:.2f}    50Hz cycles {hz50cycles:.2f}")
```

Reply:
I agree that these values seem somewhat arbitrary. As far as I can tell the register in question takes a uint8_t, so there shouldn't be a reason to restrict it to an enum.

Would you like to take a crack at updating the code to take a uint8_t instead of `tcs34725IntegrationTime_t`?

Reply:
Sure!  Maybe I'll combine the code from my other pull request (#38) into this one, it seems like it should be part of that?

Reply:
I agree, that would make sense.

Reply:
It's been a while since I've done C++ and I have a question:  I'm guessing having the helpful tags like `TCS34725_INTEGRATIONTIME_2_4MS` are nice, even if we don't need to have them be part of an enumerated type.  After making the ATIME register an int, should I make them `#define`s?  Or should I make them `const`s?  It seems nice to have them around, but I'm not sure what's the good C++ style way of doing it

Reply:
C++ isn't the sharpest tool in my toolbox either so I can't speak to there being a good/better way to do it, but I think the most important thing is that the documentation (inline and examples) show how to use the method and how different argument values will change the behavior.

I agree that the tags are helpful to make the abstract LSB value understandable, however I don't have a strong opinion about how much effort should be put into making them available. The important thing as above is that the behavior is well documented and understandable.
I'm happy with either of these options:
* change the argument, remove the enum and replace the entries with corrected macros as discussed
* Change the argument type and the argument (and possibly method) name to be the number of additional integration cycles. The docs for the argument would mention that the argument is adding additional 2.4ms integration cycles to the default of 1

My preference would be for the later but either is fine. In either case I think `setIntegrationTime` or its successor deserves its own example.

Reply:
I picked the former.  They both have a little bit of weirdness (256-value) or (value+1).  I went with the one that matches the datasheet so that at least the weirdness is consistent with the documentation.  In addition there are copious `#defines` so if you just want to set it to a constant they are all documented for you.